homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
 SELECT first_name, last_name, salary, job_title
 FROM employees
          LEFT JOIN jobs USING (job_id);
+
 -- 2. the first and last name, department, city, and state province for each employee.
+SELECT first_name, last_name, department_name, city, state_province
+FROM employees
+    LEFT JOIN departments USING (department_id)
+    LEFT JOIN locations USING (location_id);
+
 -- 3. the first name, last name, department number and department name, for all employees for departments 80 or 40.
+SELECT first_name, last_name, department_name, department_id
+FROM employees
+    LEFT JOIN departments USING (department_id)
+WHERE department_id = 80 or department_id = 40;
+
 -- 4. those employees who contain a letter z to their first name and also display their last name, department, city, and state province.
+SELECT first_name, last_name, department_name, city, state_province
+FROM employees
+    LEFT JOIN departments USING (department_id)
+    LEFT JOIN locations USING (location_id)
+WHERE first_name LIKE '%z%'
+;
 -- 5. the first and last name and salary for those employees who earn less than the employee earn whose number is 182.
+SELECT first_name, last_name, salary
+from employees
+WHERE salary < (
+    SELECT salary
+    FROM employees
+    WHERE employee_id = 182
+    );
+
 -- 6. the first name of all employees including the first name of their manager.
+SELECT e1.first_name, e2.first_name AS manager_name
+FROM employees e1, employees e2
+WHERE e2.employee_ID = e1.manager_ID;
```

Comment:
That works too :), but you can use self join, smth like:
```suggestion
-- 6. the first name of all employees including the first name of their manager.
SELECT emp.first_name as EmployeeFirst_Name, man.first_name as ManagerFirst_Name
FROM employees emp
         JOIN employees man ON (man.employee_id = emp.manager_id);
```

Reply:
Refactored to JOIN